### PR TITLE
Fix javadoc under Java 11, Fix #334

### LINKED ISSUE #334: maven-javadoc-plugin fails under Java11
```
[INFO] --- maven-javadoc-plugin:3.3.0:jar (attach-javadocs) @ chronicle-map ---
[ERROR] RuntimeException: Error while generating Javadoc: Unable to derive module descriptor for /Users/pemi/.m2/repository/org/ops4j/pax/url/pax-url-aether/2.6.2/pax-url-aether-2.6.2.jar
java.lang.module.FindException: Unable to derive module descriptor for /Users/pemi/.m2/repository/org/ops4j/pax/url/pax-url-aether/2.6.2/pax-url-aether-2.6.2.jar
    at jdk.internal.module.ModulePath.readJar (ModulePath.java:648)
    at jdk.internal.module.ModulePath.readModule (ModulePath.java:331)
    at jdk.internal.module.ModulePath.scan (ModulePath.java:237)
    at jdk.internal.module.ModulePath.scanNextEntry (ModulePath.java:190)
    at jdk.internal.module.ModulePath.findAll (ModulePath.java:166)
    at org.codehaus.plexus.languages.java.jpms.CmdModuleNameExtractor.getModuleName (CmdModuleNameExtractor.java:90)
    at org.codehaus.plexus.languages.java.jpms.LocationManager$1.extract (LocationManager.java:130)
    at org.codehaus.plexus.languages.java.jpms.LocationManager.resolvePath (LocationManager.java:379)
    at org.codehaus.plexus.languages.java.jpms.LocationManager.resolvePath (LocationManager.java:135)
    at org.apache.maven.plugins.javadoc.AbstractJavadocMojo.addJavadocOptions (AbstractJavadocMojo.java:5345)
    at org.apache.maven.plugins.javadoc.AbstractJavadocMojo.executeReport (AbstractJavadocMojo.java:2182)
    at org.apache.maven.plugins.javadoc.JavadocJar.doExecute (JavadocJar.java:199)
    at org.apache.maven.plugins.javadoc.AbstractJavadocMojo.execute (AbstractJavadocMojo.java:2010)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: java.lang.module.InvalidModuleDescriptorException: Provider class org.apache.commons.logging.impl.SLF4JLogFactory not in module
    at jdk.internal.module.ModulePath.deriveModuleDescriptor (ModulePath.java:555)
    at jdk.internal.module.ModulePath.readJar (ModulePath.java:644)
    at jdk.internal.module.ModulePath.readModule (ModulePath.java:331)
    at jdk.internal.module.ModulePath.scan (ModulePath.java:237)
    at jdk.internal.module.ModulePath.scanNextEntry (ModulePath.java:190)
    at jdk.internal.module.ModulePath.findAll (ModulePath.java:166)
    at org.codehaus.plexus.languages.java.jpms.CmdModuleNameExtractor.getModuleName (CmdModuleNameExtractor.java:90)
    at org.codehaus.plexus.languages.java.jpms.LocationManager$1.extract (LocationManager.java:130)
    at org.codehaus.plexus.languages.java.jpms.LocationManager.resolvePath (LocationManager.java:379)
    at org.codehaus.plexus.languages.java.jpms.LocationManager.resolvePath (LocationManager.java:135)
    at org.apache.maven.plugins.javadoc.AbstractJavadocMojo.addJavadocOptions (AbstractJavadocMojo.java:5345)
    at org.apache.maven.plugins.javadoc.AbstractJavadocMojo.executeReport (AbstractJavadocMojo.java:2182)
    at org.apache.maven.plugins.javadoc.JavadocJar.doExecute (JavadocJar.java:199)
    at org.apache.maven.plugins.javadoc.AbstractJavadocMojo.execute (AbstractJavadocMojo.java:2010)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[INFO] Building jar: /Users/pemi/git/chronicle/opensource/Chronicle-Map/target/chronicle-map-3.22ea3-SNAPSHOT-javadoc.jar
```

### PR DESCRIPTION
**Pull request by minborg**

### CODE REVIEW

**Review by RobAustin — COMMENTED** [review]
can this version be moved into the third-party pom

**Comment by minborg** [comment]
Plugin versions are usually in the root-parent-pom

**Comment by RobAustin** [comment]
> Plugin versions are usually in the root-parent-pom

mm, we should change that to a third party bom, but for now, for consistency should the version not be set there?

**Comment by minborg** [comment]
This did not fix the problem as intended. Building only javadoc works but not if testing was made before. So, back to the drawingboard

File: pom.xml [hunk 1]
```diff
@@ -314,6 +314,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
+                <version>3.3.0</version>
```
Comment (JerryShea) [inline pom.xml:317]: I made this same change to C-Q-E-Demo

**Pull request closed without merging**